app/models/student_membership.rb

```diff
@@ -82,4 +84,14 @@ def one_accepted_per_assignment
     errors.add(:base, I18n.t('csv.memberships_not_unique'))
     throw(:abort)
   end
+
+  def test_student_membership_status
+    user.memberships.each do |membership|
```

Comment:
this validation should only care about the membership referred to by `self`, not all memberships belonging to the user.

try something like:

```ruby
errors.add(...) if user.test_student? && membership_status != STATUSES[:inviter]
```